packages/bierzo-wallet/src/store/types.ts

```diff
@@ -0,0 +1,5 @@
+import { ThunkDispatch } from 'redux-thunk';
```

Comment:
I believe this file can be deleted, and also the redux-thunk library, or am I missing something?

Reply:
But how we will control extension availability during wallet usage session? Should we set state inside the component where we make a request to the extension's data? Or we should set state only during startup and assume that nothing will happens with extension during session?

Reply:
As far as I see the app:
1. The app on startup goes to login page
2. The user logs in, and we update the store
3. The user goes to any view different than login, we use useSelector for knowing the user status, if the user is not logged, redirected to login view.
4. If you are going to perform an operation (sign a tx), first we check the `getExtensionStatus` if we receive bad request (not extension installed or not logged) we show a toast error, we update redux showing the user is not connected, and then, we redirect the app to the login page.

I believe those 4 steps cover the basic UX flow. By the way, we need to create tickets for it. Let me know if we want to do it together.

Reply:
Unfortunately there is not `getExtensionStatus` request in background script. As you can remember we discussed this, so the only possible way to get extension status is to call `GetIdentitiesRequest` on the background script. So this call will return offline/not installed state and will return `connected` state only after user will click `Accept` on share identities view of the extension. Are you sure that we should call `GetIdentitiesRequest` every time we want to perform status check. Or we should implement another routine for this in background script?
Also we do not need to check `getExtensionStatus` before performing any operation, because any operation called on the background script will return same error like `getExtensionStatus` so we just need single point where we can check background script responses and set state accordingly. That's was my idea to use Thunk.

Reply:
> Unfortunately there is not getExtensionStatus request in background script.

No, this method lives in the app, is your **encapsulated** helper to know the status of the extension (/src/communication/status/index.ts). 

> As you can remember we discussed this, so the only possible way to get extension status is to call GetIdentitiesRequest on the background script. So this call will return offline/not installed state immediately and will return connected state only after user will click Accept on share identities view of the extension.

As we discussed, the SigningServer protocol is closed, after the MVP maybe we can refactor this part. Create a ticket and put it in the "After MVP" column. So, yeah, use the "getIdentities" trick to know the extension' status. This is not a real problem because, as explained in the last point, inside /src/communication/status/index.ts the communication is well encapsulated. The day we introduce another message in the background script (because the signing server supports it, we just need to update this method, and that is all!!)

> Are you sure that we should call GetIdentitiesRequest every time we want to perform status check. Or we should implement another routine for this in background script?

No, not a requirement from my side. You proposed it, I think we can safely assume, if redux says we are connected, we have not closed the extension. So for me, before sending any signAndPost request to the extension you want to omit this check, is fine with me.

Any other doubts?

Reply:
ok, got it. Thank you!